src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -9218,6 +9267,78 @@ export class StudyViewPageStore
             return isMixedReferenceGenome(this.queriedPhysicalStudies.result);
         }
     }
+    private _clinicalEventFilters = observable.array<DataFilter>([], {
+        deep: false,
+    });
+
+    @computed get clinicalEventFilters(): DataFilter[] {
+        return this._clinicalEventFilters.slice();
+    }
+
+    @action
+    public setClinicalEventTypeFilter(filters: DataFilter[]) {
+        this._clinicalEventFilters.replace(filters);
+    }
+
+    @action
+    public resetClinicalEventTypeFilter() {
+        this._clinicalEventFilters.clear();
+    }
+
+    @computed get clinicalEventTypeFiltersRawStrings(): string[][] {
+        const filters = _.map(this.clinicalEventFilters, filterList =>
+            _.map(filterList.values, filters => filters.value)
+        );
+        return toJS(filters);
+    }
+    public readonly clinicalEventTypeCounts = remoteData({
+        await: () => [this.selectedSamples],
+        invoke: () => {
+            return internalClient.getClinicalEventTypeCountsUsingPOST({
+                studyViewFilter: this.filters,
+            });
+        },
+    });
+
+    @action.bound
+    removeClinicalEventTypeFilter(eventTypeFilterToRemove: string): void {
+        let currentClinicalEventTypeFilters =
+            toJS(this.clinicalEventFilters) || [];
+
+        const updatedClinicalEventTypeFilters = _.forEach(
```

Comment:
we should probably use map instead of forEach.  You don't typically assign from whatever is returned from forEach.  i can explain.